### ISSUE TITLE
Check that fasta sequence identifiers do not contain a pipe symbol. 

### ISSUE THREAD

**Issue by NeillGibson** [chapmanb/bcbio-nextgen#1160]
Hi,

BCBio fails currently all the way after mapping and variant calling on trying to cat a bed file whose name is based on sequence identifier containing a pipe symbol.

Linux cat only reads the bed file name up to the pipe symbol and then complains it can't find that file. 

For instance sequence identifier  >gi|aaaaaaaa|gb|bbbbbbbb| in the reference genome creates a bed file after mapping and variant calling  called gi|aaaaaaaa|gb|bbbbbbbb|*.bed

Could you add a check in the  `bcbio_setup_genome.py` script to fail if one or more sequence identifiers in the supplied fasta file include a  pipe symbol ?

For the time being I just create an extra sequence identifier without a pipe symbol. 

I am not sure if in general there is some rule against pipe symbols in sequence identifiers in reference genome fasta files but a  quick search online leads me to some general  fasta examples that have sequence identifiers with pipe symbols in the name : 
```
>gi|46452254|gb|AY585334.1| Sus scrofa cystic fibrosis transmembrane conductance regulator 
(CFTR) mRNA, complete cds
Length=4449
>gi|31563518|ref|NP_852610.1| microtubule-associated proteins 1A/1B light chain 3A isoform b 
```
http://blast.ncbi.nlm.nih.gov/blastcgihelp.shtml
https://en.wikipedia.org/wiki/FASTA_format

Thank you.

**Comment by chapmanb** [comment]
Neill;
Sorry about the issue. It looks like we haven't properly escaped these characters when writing out file names. The quick solution is, as you suggest, to sanitize the reference names yourself before feeding into bcbio. But we'd like to fix this automatically. Do you have the traceback from where it's failing at and we can put in fixes there as well as when setting up the genome? Thanks much for the report.

**Comment by NeillGibson** [comment]
Hi Brad, 

I am afraid I already deleted the output and logs of the failed run as I was in hurry to restart the analysis with a fixed reference genome. 

The issue should reproducible by adding pipe symbols to the end of the sequence identifiers of any reference genome. 

Sorry that I can't provide any specific information on the lines of code involved.

**Comment by chapmanb** [comment]
Neill;
No worries, thanks again for reporting and letting us know. I pushed fixes which explicitly disallow use of non-standard characters in contig names. bcbio will now raise and error and let you know what is wrong instead of silently failing later. I tried to fix this with escaping but quickly realized there are a lot of potential issues feeding into tools and it's better to have clean names up front. I also added support to bcbio_setup_genomes to fix these characters during preparation. Thanks again for the report.
